wheel/lib_file_analyse.py

```diff
@@ -0,0 +1,304 @@
+        magic_number = swap32(magic_number)
+    else:
+        BaseClass = ctypes.Structure
+    lib_file.seek(seek)
+    return BaseClass, magic_number
+
+
+def read_data(struct_class, lib_file):
+    return struct_class.from_buffer_copy(lib_file.read(
+                        ctypes.sizeof(struct_class)))
+
+
+def extract_macosx_min_system_version(path_to_lib):
+    with open(path_to_lib, "rb") as lib_file:
+        BaseClass, magic_number = get_base_class_and_magic_number(lib_file, 0)
+        if magic_number not in [FAT_MAGIC, FAT_MAGIC_64, MH_MAGIC, MH_MAGIC_64]:
+            return
+        if magic_number in [FAT_MAGIC, FAT_CIGAM_64]:
+            class FatHeader(BaseClass):
+                _fields_ = fat_header_fields
+            fat_header = read_data(FatHeader, lib_file)
+            if magic_number == FAT_MAGIC:
+
+                class FatArch(BaseClass):
+                    _fields_ = fat_arch_fields
+            else:
+
+                class FatArch(BaseClass):
+                    _fields_ = fat_arch_64_fields
+            fat_arch_list = [read_data(FatArch, lib_file) for _ in range(fat_header.nfat_arch)]
+
+            class SegmentBase(BaseClass):
+                _fields_ = segment_base_fields
+            versions_list = []
+            for el in fat_arch_list:
+                try:
+                    version = read_mach_header(lib_file, el.offset)
+                    if version is not None:
+                        versions_list.append(version)
+                except ValueError:
+                    pass
+            return max(versions_list)
+        else:
+            try:
+                return read_mach_header(lib_file, 0)
+            except ValueError:
+                """when some error during read library files"""
+                return None
+
+
+def read_mach_header(lib_file, seek=None):
+    """
+    This funcition parse mach-O header and extract
+    information about minimal system version
+
+    :param lib_file: reference to opened library file with pointer
+    """
+    if seek is not None:
+        lib_file.seek(seek)
+    base_class, magic_number = get_base_class_and_magic_number(lib_file)
+    arch = "32" if magic_number == MH_MAGIC else "64"
+
+    class SegmentBase(base_class):
+        _fields_ = segment_base_fields
+    if arch == "32":
+
+        class MachHeader(base_class):
+            _fields_ = mach_header_fields
+
+        class SegmentCommand(base_class):
+            _fields_ = segment_command_fields
+    else:
+
+        class MachHeader(base_class):
+            _fields_ = mach_header_fields_64
+
+        class SegmentCommand(base_class):
+            _fields_ = segment_command_fields_64
+    mach_header = read_data(MachHeader, lib_file)
+    for _i in range(mach_header.ncmds):
+        pos = lib_file.tell()
+        segment_base = read_data(SegmentBase, lib_file)
+        lib_file.seek(pos)
+        if segment_base.cmd == LC_VERSION_MIN_MACOSX:
```

Comment:
Is it guaranteed that `LC_VERSION_MIN_MACOSX` always appears before `LC_BUILD_VERSION`?

Reply:
They are disjoint. form macosx version 10.8 to 10.13 there is `LC_VERSION_MIN_MACOSX` section and since 10.14 (current) there is `LC_BUILD_VERSION` section. 

As you can see both contains return statement so if function found one of them then it finish.